### DISCUSSION TITLE
The region manager does not contain the ContentRegion

### DISCUSSION THREAD

**Discussion by ErwinDraconis** [PrismLibrary/Prism#2358] [WPF]
Hello,
Hier is my setup based on the code examples from PRISM repo : 

XAML : 

`<ContentControl Grid.Column="0" Grid.Row="0" Prism:RegionManager.RegionName="ContentRegion" />`

C# : 

`readonly IContainerExtension _container;
        readonly IRegionManager _regionManager;

        public ProductionViewModel(IContainerExtension container, IRegionManager regionManager)
        {
            _container = container;
            _regionManager = regionManager;
            loadViews();
        }

        private void loadViews()
        {
            var _upperTrommel = _container.Resolve<BTrom>();

            var _region = _regionManager.Regions["ContentRegion"];

            _region.Add(_upperTrommel);
        }`

i get this error  : 

![image](https://user-images.githubusercontent.com/42243889/109307463-fa72ce00-7840-11eb-95f6-a48bc41c9a1a.png)

Thank you.

**Comment by brianlagunas** [comment]
This is telling you that you are trying to add views to the region before it has been created/initialized. It's too early. I'm not sure how you have your app architected, but you'll need to ensure the regions exist before trying to add to them. Or just use IRegionManager.RegisterVIewWithRegion and it will add the view automatically when the region becomes available.

**Reply by ErwinDraconis** [reply]
Hi Brian, 
i have found the issue, i guess ! the contentRegion should be always defined on the shell.
in my case i put it in different usercontrol that loads other views based on some conditions . in this case the regionManager cannot find this region that exist outside of the shell.
is there any work around in this scenario ?
Thank you.

**Reply by brianlagunas** [reply]
Thats not the issue. Regions can be defined anywhere. They are not limited to the shell.

**Reply by ErwinDraconis** [reply]
Thanks Brian, you are right.
is there a way to inject data into view model when registring the RegisterViewWithRegion ? something like
`regionManager.RegisterViewWithRegion("ContentRegion", typeof(ViewA),ObjectToBeInjected);`
Thank you.
